sktime/classification/interval_based/_rise.py

```diff
@@ -374,34 +504,87 @@ def matrix_acf(x, num_cases, max_lag):
+        lag_length = length - lag
+        x1, x2 = x[:, :-lag], x[:, lag:]
+        s1 = np.sum(x1, axis=1)
+        s2 = np.sum(x2, axis=1)
+        m1 = s1 / lag_length
+        m2 = s2 / lag_length
+        s12 = np.sum(x1 * x2, axis=1)
+        ss1 = np.sum(x1 * x1, axis=1)
+        ss2 = np.sum(x2 * x2, axis=1)
+        v1 = ss1 - s1 * m1
+        v2 = ss2 - s2 * m2
+        v12 = s12 - s1 * m2
+        v1_is_zero, v2_is_zero = v1 <= 1e-9, v2 <= 1e-9
+        non_zero = ~v1_is_zero & ~v2_is_zero
+        # y[:, lag - 1] = np.sum((x1 - m1[:, None]) *
+        # (x2 - m2[:, None]), axis=1)
+        y[v1_is_zero & v2_is_zero, lag - 1] = 1  # Both zero variance,
+        # so must be 100% correlated
+        y[v1_is_zero ^ v2_is_zero, lag - 1] = 0  # One zero variance
+        # the other not
+        var = (v1 * v2)[non_zero]
+        y[non_zero, lag - 1] = v12[non_zero] / np.sqrt(var)
+    #     # y[lag - 1] = np.corrcoef(x[:, lag:], x[:, -lag])[0][1]
+    #     # if np.isnan(y[lag - 1]) or np.isinf(y[lag - 1]):
+    #     #     y[lag - 1] = 0
     return y
 
 
-def ps(x):
-    """Power spectrum transform.
-
-    Calculated using np function. It would be worth looking at ff implementation,
-    see difference in speed to java.
+def ps(x, sign=1, n=None, pad="mean"):
+    """
+    Power spectrum transform, currently calculated using np function.
+    It would be worth looking at ff implementation, see difference in speed
+    to java.
 
     Parameters
     ----------
     x : array-like shape = [interval_width]
+    sign : {-1, 1}, default = 1
+    n : int, default=None
+    pad : str or function, default='mean'
+        controls the mode of the pad function
+        see numpy.pad for more details
+        https://numpy.org/doc/stable/reference/generated/numpy.pad.html
 
     Return
     ----------
     y : array-like shape = [len(x)/2]
     """
-    fft = np.fft.fft(x)
-    fft = fft.real * fft.real + fft.imag * fft.imag
-    fft = fft[: int(len(x) / 2)]
-    return np.array(fft)
+    x_len = x.shape[-1]
+    x_is_1d = x.ndim == 1
+    # pad or slice series if length is not of power of 2 or n is specified
+    if x_len & (x_len - 1) != 0 or n:
+        # round n (or the length of x) to next power of 2
+        # when n is not specified
+        if not n:
+            n = _round_to_nearest_power_of_two(x_len)
+        # pad series up to n when n is larger otherwise slice series up to n
+        if n > x_len:
+            pad_length = (0, n - x_len) if x_is_1d else ((0, 0), (0, n - x_len))
+            x_in_power_2 = np.pad(x, pad_length, mode=pad)
+        else:
+            x_in_power_2 = x[:n] if x_is_1d else x[:, :n]
+    else:
+        x_in_power_2 = x
+    # use sign to determine inverse or normal fft
+    # using the norm in numpy fft function
+    # backward = normal fft, forward = inverse fft (divide by n after fft)
+    norm = "backward" if sign > 0 else "forward"
+    fft = np.fft.rfft(x_in_power_2, norm=norm)
```

Comment:
Already commented but a bit more specific, seems to be failing tests here due to the norm options. Error for most is "ValueError: Invalid norm value backward, should be None or "ortho".

The numpy version used by sktime is 1.19 currently, though it seems those options are valid as of 1.2.

Reply:
hi @whackteachers Im not sure when we will update or indeed what our schedule for doing so is, but it is probably easier to just make it compatible with 1.19. Otherwise it all looks good to me

Reply:
I have made the change, plz check if I m doing it the right way